src/functions.php

```diff
@@ -227,3 +227,32 @@ function percentile($arr, $percentile = 0.95)
     sort($arr);
     return $arr[max(round($percentile * count($arr) - 1.0 - $percentile), 0)];
 }
+
+/**
+ * Download the given URI and decode it as JSON.
+ * @param string $uri
+ *
+ * @return mixed
+ */
+function getURIContentAsJson($uri){
+    // get the json file
+    $httpsProxy = getenv('https_proxy');
+    if (!empty($httpsProxy)) {
+        return json_decode(@file_get_contents($uri, false, [
```

Comment:
Style is really bad here, but this is not the most important.

Current snippet is incorrect as `file_get_contents`'s 3rd parameter must be a `resource`, not an `array`.
You miss the usage of `stream_context_create` PHP internal function to create the resource based on the parameters you want to set here.